### ISSUE TITLE
Faction info

### ISSUE THREAD

**Issue by buedefixe** [ti-assistant/issues#54]
Apologies if these issues should be separated, but they go together a bit in my addled mind.

In the faction view, the Game id box is mostly covering up the first player icon and the title covers up the unlock for Advance to Status Phase. And the Update box can overlap with techs and/or planets if there are enough.

![image](https://github.com/user-attachments/assets/0e40cb70-6930-47a3-a59f-21c8c8edac68)

As a side note, I really like the little circles to indicate which specific techs have been researched that you use on the Techs grouped by Factions page and I'd love to see them supplementing the color count on the faction pages and the main page (I get that space is at a premium on the main page, but personally I'd be willing to lose the number or lose the tech type icon to instead have easy info about which techs exactly each person has).

![image](https://github.com/user-attachments/assets/3ffb3eec-73f5-4d63-8ce4-d982833eab68)

That said, I think there's an error in the above. It seems that when there is a faction unit tech that has been researched, it's not highlighting the dot. In the above, Argent has researched SWAII, but the corresponding dot isn't lit up.

Also, while the organtization of dots for the color techs make sense to me, the unit upgrade dots are harder to understand at a glance. Instead of having a single long row organized alphabetically underneath "Upgrades", I'd suggest an array of dots that matches the unit organization on the faction sheets. Something like the below to the right of "Upgrades" (or even instead of the word "Upgrades" on the main page if you wanted to put it there without taking extra space):

○○
○○○
○○○○

Corresponding to
War Sun              Cruiser II
Dread II               Destroyer II       PDS II
Carrier II              Figher II             Infantry II         Space Dock II

For the faction pop-ups, I think it would be nice to be able to see what a faction's commander and/or hero powers are without actually unlocking them (and risking forgetting to re-lock them after looking, which could lead another player to think something is unlocked that isn't).

I haven't checked exhaustively, but I think overall, it would more effectively use screen real estate to put Leaders and Promissory Note in the same column and Abilities and Faction Techs in the same column.

I know I'm sending a lot. Thanks for listening :)

**Comment by SephHawkins** [comment]
No worries on posting issues together - if I need to track them separately for any reason, I'll just split them out. And thanks again for all the feedback, I really appreciate it.

> In the faction view, the Game id box is mostly covering up the first player icon and the title covers up the unlock for Advance to Status Phase. And the Update box can overlap with techs and/or planets if there are enough.

Yeah, the Faction page is a bit of a mess if you aren't on mobile. Originally, I thought that players would use their own faction page on their phones, but in practice it doesn't seem like a lot of groups do that. I'll take a pass on cleaning it up a bit - should just be a matter of fixing up the margins and such to make sure it doesn't overlap.

> As a side note, I really like the little circles to indicate which specific techs have been researched that you use on the Techs grouped by Factions page and I'd love to see them supplementing the color count on the faction pages and the main page (I get that space is at a premium on the main page, but personally I'd be willing to lose the number or lose the tech type icon to instead have easy info about which techs exactly each person has).

I went back and forth a lot on this - I actually added them into the Tech Summary section, but when I asked my group for feedback, they all said that it looked way too cluttered so I pulled them out. I'm planning on adding some more fine-grained controls for managing the app's UI, so when I do that I'll make sure that there's an option for displaying the Tech Trees in the summary.

> That said, I think there's an error in the above. It seems that when there is a faction unit tech that has been researched, it's not highlighting the dot. In the above, Argent has researched SWAII, but the corresponding dot isn't lit up.

That does look like a bug, but I can't seem to reproduce it - would you mind commenting the game ID so that I can dig into it?

> Also, while the organtization of dots for the color techs make sense to me, the unit upgrade dots are harder to understand at a glance. Instead of having a single long row organized alphabetically underneath "Upgrades", I'd suggest an array of dots that matches the unit organization on the faction sheets. Something like the below to the right of "Upgrades" (or even instead of the word "Upgrades" on the main page if you wanted to put it there without taking extra space):

That could work, and is probably better than the alphabetical row. I think it's still fairly hard to quickly look at and know what Upgrades a faction has though - if you've got the faction sheet in front of you that helps, but you still may need to look back and forth multiple times. There's probably a solution that does both - I just need to find it. I might try putting letters next to each, like how ti4battle.com represents them, but I need to mess around with it more to get it to look good.

> For the faction pop-ups, I think it would be nice to be able to see what a faction's commander and/or hero powers are without actually unlocking them (and risking forgetting to re-lock them after looking, which could lead another player to think something is unlocked that isn't).

Good point - I'll update those to have both the unlock and the leader text.

> I haven't checked exhaustively, but I think overall, it would more effectively use screen real estate to put Leaders and Promissory Note in the same column and Abilities and Faction Techs in the same column.

It might be. The tricky 3 are Nomad (3 Agents), Empyrean (2 Promissories) and Titans (4 Unique Units w/ lots of text). I'll have to look around at things - it might be possible to combine Faction Techs with Unique Units so that both of those can be in the right column even for factions like Titans.

**Comment by buedefixe** [comment]
The game ID with the faction unit tech bubble bug is d4tF4E. Note that most (maybe all) tech in this one was added from the faction screen, rather than from the main screen during a tech action. Also, I did a lot of playing around with things and undoing back and forth as I was putting together the feedback, so it's not surprising if I caused something particularly quirky to go wrong in it.

Yeah, it's tricky to have anything that summarizes a complex tech tree in a way that is space efficient and easy to read at a glance. No solution will ever be perfect, but I really like the little tech bubble trees even though they likewise required a bit of checking against a reference at first for me (I'm way closer to having memorized the unit layout on the faction sheet than the whole tech tree). I found that once I understood the organizing logic of the tree, it was pretty easy to learn where the techs I cared about were and not need to check anymore. I think the same would be true for upgrade techs if they had a similar spatial logic, though if there's the space, letters could be helpful (also true for the color tech trees).

One way to maybe help with the units without adding to the space used would be to use different shapes for the unit techs in addition to matching the spatial organization of the faction sheet. They don't need to be perfect icons for the units; I think they could actually be pretty simple and relatively arbitrary. It would still require people to check their faction sheet, but I think it might make it easier to not have to check more than once or twice. I'm sure it could be better optimized, but something like this:

⬡▷
⭔D△
▢◇▽○

Thanks again for all your work on this app!
